### PR TITLE
feat,fix: respect EXTISM_ENABLE_WASI_OUTPUT

### LINKED ISSUE #43: Add support for enabling WASI output via EXTISM_ENABLE_WASI_OUTPUT env var
A la the libextism sdk.

Given that `process.env` isn't evenly supported across all targets – browsers don't have an environment! – it might be best to integrate this flag in the [`InternalConfig` interface](https://github.com/extism/js-sdk/blob/main/src/interfaces.ts#L171-L180) and populate it during [config normalization](https://github.com/extism/js-sdk/blob/main/src/mod.ts#L70-L77). We can implement that with `process.env`, then compile it out to `false` in the browser builds ([here](https://github.com/extism/js-sdk/blob/main/justfile#L218-L238) and [here](https://github.com/extism/js-sdk/blob/main/justfile#L130-L143))

That flag will need to be threaded through to the [`loadWasi` call](https://github.com/extism/js-sdk/blob/main/src/foreground-plugin.ts#L157), and then through to the polyfills:

- [`./src/polyfills/node-wasi.ts`](https://github.com/extism/js-sdk/blob/926fc10095b5c37dd9a9d2a0369d6ad92787abf9/src/polyfills/node-wasi.ts#L5-L8): I think Node currently _defaults_ to stdout being [enabled via wasi](https://nodejs.org/api/wasi.html#new-wasioptions) – so in this case we'd want to gate that on whether or not the env flag is set.
- [`./src/polyfills/deno-wasi.ts`](https://github.com/extism/js-sdk/blob/main/src/polyfills/deno-wasi.ts): You'll have to play with this one a bit – the [docs are a bit spare](https://deno.land/std@0.200.0/wasi/snapshot_preview1.ts). Notably, Deno [deprecated and removed](https://github.com/denoland/deno_std/pull/3732) WASI in later versions, so if this takes too much work, don't sweat it – we'll want to look into a replacement in the near future in any case.
- [`./src/polyfills/browser-wasi.ts`](https://github.com/extism/js-sdk/blob/main/src/polyfills/browser-wasi.ts#L8-L10): If the flag is enabled, we can replace these lines with something like [this example](https://github.com/bjorn3/browser_wasi_shim/blob/main/examples/rustc.html#L43-L77).

I'd be happy to pair on this, too! This would be a great way to get your fingerprints on the js-sdk.

### PR DESCRIPTION
**Pull request by chrisdickinson**
I implemented this as an `ExtismPluginOption`, `enableWasiOutput`, that defaults to checking the environment variable (on platforms that support runtime environment variables.)

- There was a WASI browser polyfill bug with the wasistdout module. Since `wasistdout` doesn't export a `initialize` or `start` method, we weren't initializing the context appropriately. This now fixed.
- Neither Deno nor Node squelched WASI output before this commit. Now they do by default -- by opening /dev/null (or NUL on Windows) and sending that FD in instead.
- And now the browser supports WASI output as well!

Fixes #43.

### CODE REVIEW

File: src/interfaces.ts [hunk 1]
```diff
@@ -163,6 +163,13 @@ export interface ExtismPluginOptions {
   functions?: { [key: string]: { [key: string]: (callContext: CallContext, ...args: any[]) => any } } | undefined;
   allowedPaths?: { [key: string]: string } | undefined;
   allowedHosts?: string[] | undefined;
+
+  /**
+   * Whether WASI stdout should be forwarded to the host.
+   *
+   * Overrides the `EXTISM_ENABLE_WASI_OUTPUT` environment variable.
+   */
+  enableWasiOutput?: boolean | undefined;
```
Comment (nilslice) [inline src/interfaces.ts:172]: just curious, are the `?` suffixes necessary when the type definition is `x | undefined`?

File: src/mod.test.ts [hunk 2]
```diff
@@ -538,6 +538,48 @@ if (typeof WebAssembly === 'undefined') {
...
+      if ((globalThis as unknown as any).process) {
+        (
+          globalThis as unknown as Record<string, { env: Record<string, string> }>
+        ).process.env.EXTISM_ENABLE_WASI_OUTPUT = '';
+      } else if ((globalThis as unknown as any).Deno) {
+        globalThis.Deno.env.set('EXTISM_ENABLE_WASI_OUTPUT', '');
+      }
+      const plugin = await createPlugin('http://localhost:8124/wasm/wasistdout.wasm', {
+        useWasi: true,
+        enableWasiOutput: true,
```
Comment (nilslice) [inline src/mod.test.ts:574]: just to confirm, to get WASI output the host user needs to set `EXTISM_ENABLE_WASI_OUTPUT` and include `enableWasiOutput: true` here?

Comment (chrisdickinson) [reply src/mod.test.ts:574]: Ah, not quite! The option is more specific than the env var. The env var is consulted when the option isn't specifically provided during plugin creation. If the env var is set to true, a user can still explicitly squelch output by passing `enableWasiOutput: false` to a specific `createPlugin` call, & vice versa.

(I added the option primarily for the browser target: env vars are usually resolved there furing bundle time, but may require bundler plugins to do so. I wanted to ensure we're as bundler-agnostic as possible.)

Comment (nilslice) [reply src/mod.test.ts:574]: oh right, great call to support the browser env! thank you

**Review by nilslice — APPROVED** [review]
LGTM!

File: src/interfaces.ts (hunk 1 above)
Comment (chrisdickinson) [reply src/interfaces.ts]: Nope! Good catch, thanks! – removed those.

**Pull request merged**